frontend/src/metabase/admin/databases/components/DeleteDatabaseModel/DeleteDatabaseModal.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { FormEvent, useState } from "react";
+import { jt, t } from "ttag";
+
+import Button from "metabase/core/components/Button";
+import ModalContent from "metabase/components/ModalContent";
+import type { DatabaseUsageInfo } from "metabase-types/api";
+import useFetch from "metabase/hooks/use-fetch";
+import Alert from "metabase/core/components/Alert";
+import Input from "metabase/core/components/Input";
+import type Database from "metabase-lib/metadata/Database";
+import ContentRemovalConfirmation from "../ContentRemovalConfirmation";
+import {
+  DeleteDatabaseModalFooter,
+  DeleteDatabaseModalRoot,
+  DeleteDatabaseModalSection,
+  ErrorMessage,
+} from "./DeleteDatabaseModal.styled";
+
+const hasContentInDatabase = (usageInfo: DatabaseUsageInfo) => {
+  return (
+    usageInfo.question > 0 ||
+    usageInfo.dataset > 0 ||
+    usageInfo.metric > 0 ||
+    usageInfo.segment > 0
+  );
+};
+
+const getErrorMessage = (error: any) => {
+  if (!error) {
+    return null;
+  }
+
+  let errorMessage = t`Server error encountered`;
+  if (error.data && error.data.message) {
+    errorMessage = error.data.message;
+  } else if (error.message) {
+    errorMessage = error.message;
+  }
+
+  return errorMessage;
+};
+
+export interface DeleteDatabaseModalProps {
+  onClose: () => void;
+  onDelete: (database: Database) => void;
+  database: Database;
+}
+
+const DeleteDatabaseModal = ({
+  database,
+  onClose,
+  onDelete,
+}: DeleteDatabaseModalProps) => {
+  const { data: usageInfo } = useFetch<DatabaseUsageInfo>(
```

Comment:
Interesting addition of a hook around fetch here.

Random thoughts that flashed by when reading this:

Do you see this use expanding? If so, do you know where?
Should we advertise it in the FE guild?

Should we consider using a library for this? react-query comes to mind

Reply:
I'll chat about this on our FE sync. I did not add anything like react-query for now on purpose to prevent possible usage of it in different use cases since it is more powerful and provides a document cache which is controversial to our own normalized cache made with Redux. In this particular case, I needed to safely fetch these stats one-time locally for a modal — it is not a mutation, and the data won't be used anywhere else, so the simplest approach seems to be working